pkg/hubble/filters/network_interface.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package filters
+
+import (
+	"context"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+func filterByNetworkInterface(ifaces []*flowpb.NetworkInterface) FilterFunc {
+	return func(ev *v1.Event) bool {
+		iface := ev.GetFlow().GetInterface()
+		if iface == nil {
+			return false
+		}
+		for _, f := range ifaces {
+			if idx := f.GetIndex(); idx > 0 && idx != iface.GetIndex() {
```

Comment:
interface index is never 0, correct?

proto is very old at this point, but we probably missed a chance to make it a pointer to int to distinguish between 0 and nil.

Reply:
> interface index is never 0, correct?

`if_nametoindex(3)` returns `0` on error, so that was my assumption when writing the patch.

Reply:
Thanks @msune who pointed out that an iface index cannot be 0 in order to be compliant with the [SNMP RFC](https://datatracker.ietf.org/doc/html/rfc2863#section-3.1.5), see also https://github.com/cilium/cilium/pull/32526.